src/common/snippets/src/op/subgraph.cpp

```diff
@@ -210,13 +211,24 @@ Shape snippets::op::Subgraph::canonicalize(const BlockedShapeVector& outputShape
 void snippets::op::Subgraph::convert_to_snippet_dialect() {
     INTERNAL_OP_SCOPE(Subgraph);
     OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::convert_to_snippet_dialect")
+    auto skip_matching_domain = [](const std::shared_ptr<const ov::Node>& n) -> bool {
+        return n->get_input_shape(0).back() != 1;
+    };
     ngraph::pass::Manager manager;
     manager.register_pass<snippets::pass::ConvertConstantsToScalars>();
     manager.register_pass<snippets::pass::ConvertPowerToPowerStatic>();
     manager.register_pass<snippets::pass::InsertLoad>();
     manager.register_pass<snippets::pass::InsertStore>();
     manager.register_pass<snippets::pass::InsertMoveBroadcast>();
     manager.register_pass<snippets::pass::LoadMoveBroadcastToBroadcastLoad>();
+    manager.register_pass<snippets::pass::ReplaceLoadsWithScalarLoads>();
+    manager.register_pass<snippets::pass::ReplaceStoresWithScalarStores>();
+    if (exec_domain.back() != 1) {
```

Comment:
I pretty sure we need to cover this logic by comment. It is not obvious why we are replacing Loads with Scalar ones and why we don't do it for unit exec domain. 
Please fix in separate PR

Reply:
Done, added to the Snippets test infrastructure [PR](https://github.com/openvinotoolkit/openvino/pull/10605)